src/jank/clojure/core.jank

```diff
@@ -431,6 +431,92 @@
                  #{ o }#
                );"))
 
+(defn conj!
+  ([]
+   (transient []))
+  ([coll]
+   coll)
+  ([coll x]
+   (native/raw "__value = visit_object
+                (
+                  [=](auto const typed_t, auto const head) -> object_ptr
```

Comment:
Looks like these were changed from `[&]` to `[=]`, but I think they can be `[]`. Did you hit an issue with that?

Reply:
Yes, the compiler's mad at me:
```cpp
➜  jank git:(transient-vector-fns) ✗ ./bin/compile
[1/1] Generating classes/core-libraries
input_line_94:18:29: error: variable 'call_574' cannot be implicitly captured in a lambda with no capture-default specified
                    { throw call_574; }
                            ^
input_line_94:9:243: note: 'call_574' declared here
        jank::profile::timer __timer{ "dissoc_BANG__558" };auto const map_575(jank::make_box<jank::runtime::obj::persistent_array_map>(jank::runtime::detail::in_place_unique{}, jank::make_array_box<object_ptr>(const_561, coll),2));auto const call_574(jank::runtime::dynamic_call(ex_info_559->deref(), const_560, map_575));object_ptr native_573{ obj::nil::nil_const() };{ object_ptr __value{ obj::nil::nil_const() };__value = visit_object
                                                                                                                                                                                                                                                  ^
input_line_94:11:19: note: lambda expression begins here
                  [](auto const typed_t, auto const key) -> object_ptr
                  ^
input_line_94:11:20: note: capture 'call_574' by value
                  [](auto const typed_t, auto const key) -> object_ptr
                   ^
                   call_574
```

Reply:
Ah, that's because you're throwing `coll` instead of `typed_coll`. Since you have the fully typed version as the first param to the lambda, we can use that rather than also capturing the erased version.